File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/SouqCom/CategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoryRow } from "./CategoryRow";

export interface CategoryColumns {
    Id: Column<CategoryRow>;
    Name: Column<CategoryRow>;
    Photo: Column<CategoryRow>;
    Description: Column<CategoryRow>;
}

export class CategoryColumns extends ColumnsBase<CategoryRow> {
    static readonly columnsKey = 'SouqCom.Category';
    static readonly Fields = fieldsProxy<CategoryColumns>();
}

File: AdminPanel1/AdminPanel1.Web/Modules/SouqCom/Category/CategoryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel1.SouqCom.Columns;

[ColumnsScript("SouqCom.Category")]
[BasedOnRow(typeof(CategoryRow), CheckNames = true)]
public class CategoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}